one_fm/api/doc_events.py

```diff
@@ -244,3 +244,67 @@ def update_training_event_data(doc, method):
 				'training': doc.event_name,
 			})
 			doc_esm.save()
+
+
+def after_insert_of_payroll_entry(doc, method):
+	if doc.employees:
+		frappe.enqueue(generate_site_allowance_after_payroll_insert, payroll_entry=doc)
+
+
+def on_cancel_of_payroll_entry(doc, method):
+	frappe.enqueue(cancel_all_or_delete_additional_salary_related_to_payroll_entry, payroll_entry=doc, action="Cancel")	
+
+
+def on_delete_of_payroll_entry(doc, method):
+	frappe.enqueue(cancel_all_or_delete_additional_salary_related_to_payroll_entry, payroll_entry=doc, action="Delete")
+
+
+def generate_site_allowance_after_payroll_insert(payroll_entry):
+	#get list of all site that includes Site Allowance.
+	operations_sites = frappe.get_all("Operations Site", {"include_site_allowance":"1"},["name","allowance_amount"])
+
+	#Gather the required Date details such as start date, and respective end date. Get current year and month to get the no. of days in the current month.
+	start_date = datetime.strptime(payroll_entry.start_date, '%Y-%m-%d')
+	end_date = datetime.strptime(payroll_entry.end_date, '%Y-%m-%d')
+	currentMonth = end_date.month
+	currentYear = start_date.year
+	no_of_days = monthrange(currentYear, currentMonth)[1]
+
+
+	tup_of_employee = tuple([obj.employee for obj in payroll_entry.employees])
+	if operations_sites:
+			for site in operations_sites:
+				employee_details = frappe.db.sql("""
+    							SELECT employee, COUNT(attendance_date) as count
+    							FROM `tabAttendance`
+    							WHERE site = %s AND attendance_date BETWEEN %s AND %s AND status = %s AND employee IN %s GROUP BY employee
+								""", (site.name, start_date, end_date, "Present", tup_of_employee), as_dict=True)
+
+				if employee_details:
+					for employee in employee_details:
+							#calculate Monthly_site_allowance with the rate of allowance per day.
+							Monthly_site_allowance =  round(int(site.allowance_amount)/no_of_days, 3)*int(employee["count"])

```

Comment:
I dont know your number of days but it should be total days attended(Present for each employee, seek help from Jamsheer